collector/collector.go

```diff
@@ -375,13 +382,21 @@ func getDynamicNamespaceElements(handler *snmpgo.SNMP, results []*snmpgo.VarBind
 			if serr != nil {
 				return serr
 			}
-			for _, part := range parts {
-				metricNamePart := ns.ReplaceNotAllowedCharsInNamespacePart(part.Variable.String())
-				metric.Namespace[i].Values = append(metric.Namespace[i].Values, metricNamePart)
+
+			partsMap, serr := snmpResults2Map(parts, metric.Namespace[i].Oid)
+			if serr != nil {
+				return serr
+			}
+
+			for idx, part := range partsMap {
+				if _, ok := resultMap[idx]; ok {
```

Comment:
What if there is an element in `resultMap` for which there is not matching element in `partsMap`?